src/nemos/basis.py

```diff
@@ -815,14 +852,18 @@ def evaluate(self, sample_pts: NDArray) -> NDArray:
         if any(sample_pts < 0) or any(sample_pts > 1):
             raise ValueError("Sample points for RaisedCosine basis must lie in [0,1]!")
 
-        # transform to the proper domain
-        transform_sample_pts = self._transform_samples(sample_pts)
-
-        shifted_sample_pts = (
-            transform_sample_pts[:, None]
-            - (np.pi * np.arange(self.n_basis_funcs))[None, :]
+        peaks = np.linspace(0, 1, self.n_basis_funcs)
+        delta = peaks[1] - peaks[0]
+        basis_funcs = 0.5 * (
+            np.cos(
+                np.clip(
```

Comment:
I think I'd like some comments explaining this kind of confusing set of operations. What we're doing here is constructing a different set of shifted sampled points for each of the basis functions and constraining them to a single period of the cosine function. So saying something along that.